beam/core/src/main/java/cz/o2/proxima/beam/direct/io/CommitLogRead.java

```diff
@@ -112,10 +131,17 @@ public ProcessContinuation process(
             return ProcessContinuation.stop();
           }
           output.outputWithTimestamp(element, Instant.ofEpochMilli(element.getStamp()));
+          readContext.set(observer.getLastReadContext());
         }
       } finally {
-        observer.stop(false);
+        // nack all buffered but unprocessed data iff we have not emitted any output
```

Comment:
typo: `iff`

Reply:
`iff` if abbrebiaviation of `if and only if`.

Reply:
hmm, ok :) I'd maybe better not to use these, but I'll leave this up to you